projects/output-and-error-handling/main.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+type RequestError struct {
+	SleepTime time.Duration
+	ShouldRetry     bool
+	Cause       error
+}
+func main() {
+	c := http.Client{Timeout: time.Duration(1) * time.Second}
+
+	for i := 0; i < 3; i++ {
+		resp, err := c.Get("http://localhost:8080")
+
+		if err != nil {
+			fmt.Fprint(os.Stderr, "cannot get weather at the moment. Please try again later\n")
+			os.Exit(2)
+		}
+		defer resp.Body.Close()
+		
+		body, reqErr := parseResponse(*resp)
+
+		if reqErr.Cause != nil {
+			fmt.Fprintln(os.Stderr, reqErr.Cause)
+
+			if reqErr.ShouldRetry {
+				sleep(reqErr.SleepTime)
+				continue
+			} else {
+				os.Exit(28)
+			}
+		} else {
+			fmt.Fprintln(os.Stdout, body)
+			break
+		}
+	}
+}
+
+func parseResponse(resp http.Response) (reqBody string, reqError RequestError) {
+
+	switch resp.StatusCode {
+	case 429:
+		retryTime := resp.Header.Get("Retry-After")
+		formattedRetryTime, err := ParseRetryValue(retryTime)
```

Comment:
Formatted is an interesting prefix to this variable name - what does it mean here?